packages/protobuf-test/src/binary.test.ts

```diff
@@ -116,6 +119,20 @@ describe(`binary serialization`, () => {
     int32EnuField: { 1: 0, 2: 1, 0: 2 },
     int64EnuField: { "-1": 0, "2": 1, "0": 2 },
   });
+  test(MapsMessageBug1183Schema.typeName, () => {
+    const str128bytes = "x".repeat(128);
+    const msg = create(MapsMessageBug1183Schema, {
+      map: {
+        [str128bytes]: 1,
+      },
+    });
+    const bytes = toBinary(MapsMessageBug1183Schema, msg);
+    const msg2 = fromBinary(MapsMessageBug1183Schema, bytes);
+    expect(msg2.map).toStrictEqual({
+      [str128bytes]: 1,
+    });
```

Comment:
This expectation currently fails with:

```
      Object {
    -   "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx": 1,
    +   "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx": 0,
      }
```

The map entry message is more than 128 bytes long, which is a 2-byte varint. This causes the while-loop in `readMapEntry` to exit early, and never read the value field, falling back to the implicit default `0`.